klippy/extras/raspberry_pi_temperature.py

```diff
@@ -0,0 +1,62 @@
+# Support for Raspberry Pi temperature sensor
+#
+# Copyright (C) 2020  Al Crate <al3ph@users.noreply.github.com>
+#
+# This file may be distributed under the terms of the GNU GPLv3 license.
+
+import logging
+
+RASPBERRYPI_REPORT_TIME = 1.0
+RASPBERRYPI_MIN_REPORT_TIME = 0.5
+
+class RaspberryPiTemperature:
+    def __init__(self, config):
+        self.printer = config.get_printer()
+        self.name = config.get_name().split()[-1]
+        self.reactor = self.printer.get_reactor()
+        self.report_time = config.getint(
+            'raspberry_pi_temperature_report_time',
+            RASPBERRYPI_REPORT_TIME,
+            minval=RASPBERRYPI_MIN_REPORT_TIME
+        )
+        self.temp = 0.0
+        self.printer.add_object("raspberry_pi_temperature " + self.name, self)
+        self.sample_timer = self.reactor.register_timer(
+            self._sample_pi_temperature
+        )
+        self.printer.register_event_handler("klippy:ready", self.handle_ready)
+
+    def handle_ready(self):
+         self.reactor.update_timer(self.sample_timer, self.reactor.NOW)
+
+    def setup_minmax(self, min_temp, max_temp):
+        pass
+
+    def setup_callback(self, cb):
+        self._callback = cb
+
+    def _sample_pi_temperature(self, eventtime):
+        try:
+            f = open("/sys/class/thermal/thermal_zone0/temp", "r")
+            self.temp = float(f.read())/1000.0
+        except Exception:
+            logging.exception("raspberry_pi_temperature: Error reading data")
+            self.temp = 0.0
+            return self.reactor.NEVER
+
+        measured_time = self.reactor.monotonic()
+        self._callback(measured_time, self.temp)
+        return measured_time + self.report_time
+
+    def get_status(self, eventtime):
+        return {
+            'temperature': self.temp,
+        }
```

Comment:
FYI, as you indicated in your comments, this should be added to `docs/Command_Templates.md` (and the config parameters should be added to `docs/Config_Reference.md`).

-Kevin

Reply:
Will do.